contrib/core-contract-tests/tests/sip-031/sip-031.fast-check.test.ts

```diff
@@ -0,0 +1,223 @@
+import { project, accounts } from '../clarigen-types';
+import { projectFactory } from '@clarigen/core';
+import { rov, txErr, txOk } from '@clarigen/test';
+import { test, expect } from 'vitest';
+import * as fc from 'fast-check';
+import { Cl } from '@stacks/transactions';
+
+const contracts = projectFactory(project, 'simnet');
+const contract = contracts.sip031;
+const constants = contract.constants;
+const indirectContract = contracts.sip031Indirect;
+
+/**
+ * "Mint" STX to the contract
+ */
+function mint(amount: number | bigint) {
+  txOk(
+    indirectContract.transferStx(amount, contract.identifier),
+    accounts.wallet_4.address,
+  );
+}
+
+// Helper function to mint the initial 200M STX to the contract
+function mintInitial() {
+  // First make sure wallet_4 has enough STX to mint the initial amount
+  txOk(
+    indirectContract.transferStx(
+      constants.INITIAL_MINT_AMOUNT / 2n,
+      accounts.wallet_4.address,
+    ),
+    accounts.wallet_5.address,
+  );
+  txOk(
+    indirectContract.transferStx(
+      constants.INITIAL_MINT_AMOUNT / 2n,
+      accounts.wallet_4.address,
+    ),
+    accounts.wallet_6.address,
+  );
+  // Mint the entire INITIAL_MINT_AMOUNT to the vesting contract
+  mint(constants.INITIAL_MINT_AMOUNT);
+}
+
+function months(n: number) {
+  return n * Number(constants.INITIAL_MINT_VESTING_ITERATION_BLOCKS);
+}
+
+test('property: vesting calculations are always mathematically correct', async () => {
+  await fc.assert(
+    fc.asyncProperty(
+      fc.integer({
+        min: 0,
+        max: 100 * Number(constants.INITIAL_MINT_VESTING_ITERATION_BLOCKS),
+      }), // blocks elapsed (0-100 months)
+      fc.bigInt({ min: 1n, max: 1000000n * 1000000n }), // extra deposit (1 micro-STX to 1M STX)
+      async (blocksElapsed, extraDeposit) => {
+        const manifestPath = global.options.clarinet.manifestPath;
+        await simnet.initSession(process.cwd(), manifestPath);
```

Comment:
Why do some tests initialize simnet manually while others rely on the implicit one?